docker/generic/start_proxy.py

```diff
@@ -1515,6 +1519,21 @@ def start_envoy(args):
     t.start()
     return proc
 
+def sigterm_handler(signum, frame):
+    """ Handler for SIGTERM and SIGINT, pass the SIGTERM to all child processes. """
+    signame = signal.Signals(signum).name
+    logging.warning("{}: got signal: {}".format(datetime.utcnow().isoformat(timespec='microseconds'), signame))
+
+    global pid_list
+    for pid in pid_list:
+        logging.info("sending TERM to PID={}".format(pid))
+        try:
+            os.kill(pid, signal.SIGTERM)
+        except OSError:
+            logging.error("error sending TERM to PID={} continuing".format(pid))
+
+    logging.info("exiting...")
+    sys.exit(0)
```

Comment:
no need to exit, just let the health check on line 1553 handle this

Reply:
Don't you need to give Envoy time to clean up when it receives SIGINT? Looking through Envoy's code, it does some items like flushing stats, finishing up any pending requests in the filter chain, etc. This may take a few seconds to complete. I imagine max 5 seconds.

You are exiting directly after SIGTERM, that is not enough time.

Even if you change to use health checker, you are only giving it 2 seconds with `HEALTH_CHECK_PERIOD` change. Maybe change it to 5 seconds.

Reply:
You may follow http://shortn/_LbIgHpcwUK

Reply:
Done